### DISCUSSION TITLE
Check if two images are perceptually the same

### DISCUSSION THREAD

**Discussion by paopre** [ImageMagick/ImageMagick#5826] [Digital Image Processing]
Hello,

Is there a portable way to check if two images are perceptually the same?

I can use:

compare -metric phash img1 img2 diffimage;

... but, instead of producing an output image (diffimage), I wonder if is there a way to return sort of YES/NO from the previous command line.
Or, at least, I wonder if I can check in some way that diffimage doesn't contain colors (IIUC, red pixels indicate that images are different)

Thanks,

**Comment by snibgo** [comment]
How do you define "perceptually the same"?

For ordinary photos or video frames, I reckon two images are perceptually the same if the RMSE is less than 0.01 on a scale of 0.0 to 1.0, and any change is distributed roughly equally across the image.

Testing the RMSE:
```
magick compare -metric RMSE in1.ext in2.ext NULL:
```
I haven't formally defined "roughly equally across the image". But this...
```
magick in1.ext in2.ext -compose Difference -composite -auto-level out.ext
```
and then viewing out.ext shows me where the differences are.

**Reply by paopre** [reply]
> How do you define "perceptually the same"?
> 

there's an utility which is called "perceptualdiff"

https://github.com/myint/perceptualdiff

which compares two images using a perceptually based image metric. No need to specify additional parameters, just do:

perceptualdiff img1 img2

and you know if the images are visually the same.


> For ordinary photos or video frames, I reckon two images are perceptually the same if the RMSE is less than 0.01 on a scale of 0.0 to 1.0, and any change is distributed roughly equally across the image.
> 
> Testing the RMSE:
> 
> ```
> magick compare -metric RMSE in1.ext in2.ext NULL:
> ```

Thanks, this is what I meant. I did not know how to output the compare result as info without producing a diffimage
IIUC:

compare -metric phash img1 img2 NULL: 

is equivalent to perceptualdiff. Am I wrong?
